pkg/controller/v1alpha1/llmisvc/fixture/gwapi_builders.go

```diff
@@ -0,0 +1,364 @@
+/*
+Copyright 2025 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fixture
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+type ObjectOption[T client.Object] func(T)
+
+type GatewayOption ObjectOption[*gwapiv1.Gateway]
+
+func Gateway(name string, opts ...GatewayOption) *gwapiv1.Gateway {
+	gw := &gwapiv1.Gateway{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: gwapiv1.GatewaySpec{
+			GatewayClassName: defaultGatewayClass,
```

Comment:
there is a gatewayclass in the configmap, we can read from there. 
https://github.com/kserve/kserve/blob/master/config/configmap/inferenceservice.yaml#L293

Reply:
nvm, did not realize these are fixtures based on the name of the file.